lib/db/mysql.js

```diff
@@ -1013,10 +1027,20 @@ module.exports = function (log, error) {
   }
 
   MySql.prototype.getConnection = function (name) {
-    return retryable(
-      this.getClusterConnection.bind(this, name),
-      [ER_TOO_MANY_CONNECTIONS, 'ECONNREFUSED', 'ETIMEDOUT', 'ECONNRESET']
-    )
+    return new P((resolve, reject) => {
+      retryable(
+        this.getClusterConnection.bind(this, name),
+        [ER_TOO_MANY_CONNECTIONS, 'ECONNREFUSED', 'ETIMEDOUT', 'ECONNRESET']
+      ).then((connection) => {
+        connection.query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;', (err) => {
```

Comment:
IIUC this will execute "`SET NAMES`" for every db query, even if an existing connection is being re-used from the pool.  Over in profile-server we did a gross little hack to avoid the overhead here and only "`SET NAMES`" once per connection, it may be worth doing the same here:

  https://github.com/mozilla/fxa-profile-server/blob/master/lib/db/mysql/index.js#L236